DelvUI/Interface/AstrologianHudWindow.cs

```diff
@@ -313,52 +287,133 @@ internal class AstrologianHudWindow : HudWindow {
+                                    }
+                                    else {
+                                        redrawText = (redrawCastInfo % 30).ToString("N1");
+                                    }
+                                }
+                                else {
+                                    if (_config.ShowRedrawTextBar) {
+                                        redrawText = (redrawCastInfo % 30).ToString("N0") + " [" + redrawStacks.ToString("N0") + "]";
+                                    }
+                                    else {
+                                        redrawText = (redrawCastInfo % 30).ToString("N0");
+                                    }
+                                }
+                            }
+                            else {
+                                if (_config.ShowRedrawTextBar) {
+                                    redrawText = redrawStacks.ToString("N0");
+                                }
+                                else {
+                                    redrawText = "";
+                                }
+                            }
+                        }
+                    }
+                    else {
+                        if (_config.ShowRedrawCooldownTextBar) {
+                            if (_config.EnableDecimalRedrawBar) {
+                                if (_config.ShowRedrawTextBar) {
+                                    redrawText = redrawCastInfo.ToString("N1") + " [" + redrawStacks.ToString("N0") + "]";
+                                }
+                                else {
+                                    redrawText = redrawCastInfo.ToString("N1");
+                                }
+                            }
+                            else {
+                                if (_config.ShowRedrawTextBar) {
+                                    redrawText = redrawCastInfo.ToString("N0") + " [" + redrawStacks.ToString("N0") + "]";
+                                }
+                                else {
+                                    redrawText = redrawCastInfo.ToString("N0");
+                                }
+                            }
+
+                        }
+                        else {
+                            if (_config.ShowRedrawTextBar) {
+                                redrawText = redrawStacks.ToString("N0");
+                            }
+                            else {
+                                redrawText = "";
+                            }
+                        }
+                    }
+                }
+                else {
+                    if (_config.ShowRedrawTextBar) {
+                        redrawText = redrawStacks.ToString("N0");
+                    }
+                    else {
+                        redrawText = "";
+                    }
+                }
+
```

Comment:
Ok this is still a monster haha
I think you should refactor it into a function like this (the code below might have some errors, please double check):

```
protected string RedrawText(int redrawStacks) {
    var ShowRedrawBar = true;
    var redrawCastInfo = 0;
    var EnableRedrawCooldownCumulated = false;
    var ShowRedrawCooldownTextBar = false;
    var EnableDecimalRedrawBar = false;
    var ShowRedrawTextBar = true;

    if (!ShowRedrawBar) {
        return "";
    }

    if (redrawCastInfo <= 0) {
        return ShowRedrawTextBar ? redrawStacks.ToString("N0") : "";
    }

    if (!EnableRedrawCooldownCumulated && (redrawCastInfo % 30) == 0) {
        return "30";
    }

    if (ShowRedrawCooldownTextBar) {
        var format = EnableDecimalRedrawBar ? "N1" : "N0";
        return ShowRedrawTextBar ? (redrawCastInfo % 30).ToString(format) + " [" + redrawStacks.ToString(format) + "]" : (redrawCastInfo % 30).ToString(format);
    }

    return ShowRedrawTextBar ? redrawStacks.ToString("N0") : "";
}
```